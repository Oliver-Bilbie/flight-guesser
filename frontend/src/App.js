import React from "react";
import { Grommet, Box } from "grommet";
import Theme from "./theme";
import Banner from "./components/Banner";
import Game from "./components/Game";

function App() {
  return (
    <div className="App">
      <Grommet theme={Theme} full={true}>
        <Banner />
        <Box alignContent="center" align="center" justify="center" pad="large">
          <Game />
        </Box>
      </Grommet>
    </div>
  );
}

export default App;
