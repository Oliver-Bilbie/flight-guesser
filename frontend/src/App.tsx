import { useState, useEffect } from "react";
import Header from "./components/Header/Header";
import Footer from "./components/Footer/Footer";
import ScoreDisplay from "./components/ScoreDisplay/ScoreDisplay";
import MakeGuess from "./components/MakeGuess/MakeGuess";
import SettingsMenu from "./components/SettingsMenu/SettingsMenu";
import SettingsButton from "./components/SettingsButton/SettingsButton";
import { useThemeStore } from "./utils/themeStore";

function App() {
  const [showSettings, setShowSettings] = useState(false);

  const theme = useThemeStore((state) => state.theme);

  useEffect(() => {
    document.documentElement.classList.toggle("dark", theme === "dark");
  }, [theme]);

  return (
    <>
      <Header />
      <ScoreDisplay />

      {showSettings ? (
        <SettingsMenu onClose={() => setShowSettings(false)} />
      ) : (
        <MakeGuess />
      )}

      <SettingsButton onClick={() => setShowSettings(true)} />

      <Footer />
    </>
  );
}

export default App;
