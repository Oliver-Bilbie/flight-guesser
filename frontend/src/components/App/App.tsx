import React from "react";
import { Grommet, Box } from "grommet";
import Theme from "../../theme";
import Banner from "../Banner/Banner";
import Footer from "../Footer/Footer";
import Game from "../Game/Game";

const App: React.FC = (): React.ReactElement => {
  return (
    <div className="App">
      <Grommet theme={Theme} full={true}>
        <Banner />
        <Box alignContent="center" align="center" justify="center" pad="large">
          <Game />
        </Box>
        <Footer />
      </Grommet>
    </div>
  );
};

export default App;
