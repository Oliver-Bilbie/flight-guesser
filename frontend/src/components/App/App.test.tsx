import React from "react";
import renderer from "react-test-renderer";
import App from "./App";

test("Matches snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
