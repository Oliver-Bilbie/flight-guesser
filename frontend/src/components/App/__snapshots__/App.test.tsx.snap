// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="StyledGrommet-sc-19lkkz7-0 gNIRdc"
  >
    <header
      className="StyledBox-sc-13pk1d4-0 bPvFHc"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 exzkbd"
      >
        <h2
          className="StyledHeading-sc-1rdh4aw-0 dvCPQZ"
        >
          Flight Guesser
        </h2>
      </div>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iISQcM"
      />
      <nav
        className="StyledBox-sc-13pk1d4-0 BPjMB"
        color="text-strong"
      >
        <a
          aria-label="email"
          className="StyledAnchor-sc-1rp7lwl-0 ftDlUk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          <svg
            aria-label="MailOption"
            className="StyledIcon-sc-ofa7kd-0 blohcu"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 4h22v16H1V4zm0 1 11 8.5L23 5"
              fill="none"
              stroke="#000"
              strokeWidth="2"
            />
          </svg>
        </a>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iISQcM"
        />
        <a
          aria-label="github"
          className="StyledAnchor-sc-1rp7lwl-0 ftDlUk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          <svg
            aria-label="Github"
            className="StyledIcon-sc-ofa7kd-0 blohcu"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.999 1C5.926 1 1 5.925 1 12c0 4.86 3.152 8.983 7.523 10.437.55.102.75-.238.75-.53 0-.26-.009-.952-.014-1.87-3.06.664-3.706-1.475-3.706-1.475-.5-1.27-1.221-1.61-1.221-1.61-.999-.681.075-.668.075-.668 1.105.078 1.685 1.134 1.685 1.134.981 1.68 2.575 1.195 3.202.914.1-.71.384-1.195.698-1.47-2.442-.278-5.01-1.222-5.01-5.437 0-1.2.428-2.183 1.132-2.952-.114-.278-.491-1.397.108-2.91 0 0 .923-.297 3.025 1.127A10.536 10.536 0 0 1 12 6.32a10.49 10.49 0 0 1 2.754.37c2.1-1.424 3.022-1.128 3.022-1.128.6 1.514.223 2.633.11 2.911.705.769 1.13 1.751 1.13 2.952 0 4.226-2.572 5.156-5.022 5.428.395.34.747 1.01.747 2.037 0 1.47-.014 2.657-.014 3.017 0 .295.199.637.756.53C19.851 20.979 23 16.859 23 12c0-6.075-4.926-11-11.001-11"
              fill="#333"
              fillRule="evenodd"
            />
          </svg>
        </a>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iISQcM"
        />
        <a
          aria-label="linkedin"
          className="StyledAnchor-sc-1rp7lwl-0 ftDlUk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          <svg
            aria-label="Linkedin"
            className="StyledIcon-sc-ofa7kd-0 blohcu"
            viewBox="0 0 24 24"
          >
            <path
              d="M20.452 20.45h-3.56v-5.57c0-1.328-.022-3.036-1.85-3.036-1.851 0-2.134 1.447-2.134 2.942v5.664H9.352V8.997h3.413v1.566h.049c.475-.9 1.636-1.85 3.367-1.85 3.605 0 4.27 2.371 4.27 5.456v6.281zM5.339 7.433a2.063 2.063 0 1 1 0-4.13 2.065 2.065 0 0 1 0 4.13zM7.12 20.45H3.558V8.997H7.12V20.45zM23 0H1a1 1 0 0 0-1 1v22a1 1 0 0 0 1 1h22a1 1 0 0 0 1-1V1a1 1 0 0 0-1-1z"
              fill="#0077B5"
              fillRule="evenodd"
            />
          </svg>
        </a>
      </nav>
    </header>
    <div
      className="StyledBox-sc-13pk1d4-0 vYUEo"
    >
      <div
        className="StyledStack-sc-ajspsk-0 RUPca"
      >
        <div
          className="StyledStack__StyledStackLayer-sc-ajspsk-1 jvatDH"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 eCzPCY"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 kQWYGB"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 gaHiVX"
              />
            </div>
            <button
              className="StyledButton-sc-323bzc-0 gPymKT"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-label="Performance"
                className="StyledIcon-sc-ofa7kd-0 jQvEgu"
                viewBox="0 0 24 24"
              >
                <path
                  d="m12 19-2 3-3-1-.5-3.5L3 17l-1-3 3-2-3-2 1-3 3.5-.5L7 3l3-1 2 3 2-3 3 1 .5 3.5L21 7l1 3-3 2 3 2-1 3-3.5.5L17 21l-3 1-2-3zm0-3a4 4 0 1 0 0-8 4 4 0 0 0 0 8z"
                  fill="none"
                  stroke="#000"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 hbIWsY"
    >
      <span
        className="StyledText-sc-1sadyjn-0 jbJJSg"
      >
        Version: 
      </span>
      <a
        aria-label="changelog"
        className="StyledAnchor-sc-1rp7lwl-0 cQsqtn"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        What's new?
      </a>
    </div>
  </div>
</div>
`;
